File: PA5.java
class Solution {
    public int search(int[] nums, int target) {
        if(nums.length ==0){
            return -1;
        }
        
        int i=0;
        int j= nums.length-1;
        
        int mid=0;
        while(i<=j){
            mid= (i+j)/2;
            
            if(nums[mid]== target){
                return mid;
            }
           if(nums[mid]>=nums[i]){
               if(target>=nums[i] && target<nums[mid]){
                   j= mid-1;
               }
               else{
                   i=mid+1;
               }
           }
            else{
                if(target>nums[mid] && target<=nums[j]){
                   i= mid+1;
               }
               else{
                   j=mid-1;
               }
            }
        
        }
        return -1;
    }
}

File: PA_5.py
#Leetcode 33 : Search in Rotated Sorted Array
#Time Complexity : O(logn)
#Space Complexity : O(1)
class Solution(object):
    def search(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        if len(nums) == 0:
            return -1 
        start = 0
        end = len(nums) - 1
        while start <= end:
            mid = (start + end) // 2
            #print(start,mid,end)
            if nums[mid] == target:
                return mid
            elif nums[mid] >= nums[start]:
                if nums[start] <= target < nums[mid]:
                    end = mid - 1
                else:
                    start = mid + 1
            else:
                if nums[mid] < target <= nums[end]:
                    start = mid + 1
                else:
                    end = mid - 1
        
        return -1